### PR TITLE
Add egressMTLS docs

### PR DESCRIPTION
**Pull request by lucacome**
Adding docs for Egress MTLS

### CODE REVIEW

File: docs-web/configuration/policy-resource.md [hunk 1]
```diff
@@ -335,6 +341,77 @@ policies:
...
+   * - ``tlsSecret``
+     - The name of the Kubernetes secret that stores the TLS certificate and key. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``tls.crt`` and ``tls.key``, otherwise the secret will be rejected as invalid.
+     - ``string``
+     - No
+   * - ``trustedCertSecret``
+     - The name of the Kubernetes secret that stores the CA certificate. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``ca.crt``, otherwise the secret will be rejected as invalid.
+     - ``string``
+     - No
+   * - ``verifyServer``
+     - Verification of the upstream HTTPS server certificate.
```
Comment (Dean-Coakley) [inline docs-web/configuration/policy-resource.md:376]: Optional:
```suggestion
     - Enables verification of the upstream HTTPS server certificate.
```

**Review by Dean-Coakley — APPROVED** [review]

**Review by LorcanMcVeigh — APPROVED** [review]

File: docs-web/configuration/policy-resource.md [hunk 2]
```diff
@@ -335,6 +341,77 @@ policies:
...
+     - Required
+   * - ``tlsSecret``
+     - The name of the Kubernetes secret that stores the TLS certificate and key. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``tls.crt`` and ``tls.key``, otherwise the secret will be rejected as invalid.
+     - ``string``
+     - No
+   * - ``trustedCertSecret``
+     - The name of the Kubernetes secret that stores the CA certificate. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``ca.crt``, otherwise the secret will be rejected as invalid.
+     - ``string``
+     - No
+   * - ``verifyServer``
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:375]: the types in the type column are displayed incorrectly:

image

File: docs-web/configuration/policy-resource.md [hunk 3]
```diff
@@ -335,6 +341,77 @@ policies:
...
+     - Enables verification of the upstream HTTPS server certificate.
+     - ``bool`
+     - No
+   * - ``verifyDepth``
+     - Sets the verification depth in the client certificates chain. The default is ``1``.
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:380]:
```suggestion
     - Sets the verification depth in the proxied HTTPS server certificates chain. The default is ``1``.
```

File: docs-web/configuration/policy-resource.md [hunk 4]
```diff
@@ -335,6 +341,77 @@ policies:
...
+   * - ``tlsSecret``
+     - The name of the Kubernetes secret that stores the TLS certificate and key. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``tls.crt`` and ``tls.key``, otherwise the secret will be rejected as invalid.
+     - ``string``
+     - No
+   * - ``trustedCertSecret``
+     - The name of the Kubernetes secret that stores the CA certificate. It must be in the same namespace as the Policy resource. The certificate must be stored in the secret under the key ``ca.crt``, otherwise the secret will be rejected as invalid.
+     - ``string``
+     - No
+   * - ``verifyServer``
+     - Enables verification of the upstream HTTPS server certificate.
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:376]: the default is missing

File: docs-web/configuration/policy-resource.md [hunk 5]
```diff
@@ -335,6 +341,77 @@ policies:
...
+     - No
+   * - ``verifyDepth``
+     - Sets the verification depth in the client certificates chain. The default is ``1``.
+     - ``int``
+     - No
+   * - ``sessionReuse``
+     - Enables reuse of SSL sessions to the upstreams. The default is ``true``.
+     - ``bool`
+     - No
+   * - ``serverName``
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:387]: the default is missing

File: docs-web/configuration/policy-resource.md [hunk 6]
```diff
@@ -335,6 +341,77 @@ policies:
...
+     - Enables passing of the server name through ``Server Name Indication`` extension.
+     - ``bool`
+     - No
+   * - ``sslName``
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:391]: the default is missing.

File: docs-web/configuration/policy-resource.md [hunk 7]
```diff
@@ -335,6 +341,77 @@ policies:
...
+     - Allows overriding the server name used to verify the certificate of the upstream HTTPS server.
+     - ``string`
+     - No
+   * - ``ciphers``
+     - Specifies the enabled ciphers for requests to an upstream HTTPS server. The default is ``DEFAULT``.
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:396]: we need to put a note here that explains that this and the field below aren't validated. How about this?

Note: the value is not validated by the Ingress Controller. As a result, NGINX can fail to reload the configuration. To ensure that the configuration for a VirtualServer/VirtualServerRoute that references the policy was successfully applied, check its [status](https://deploy-preview-1186--nginx-kubernetes-ingress.netlify.app/nginx-ingress-controller/configuration/global-configuration/reporting-resources-status/#virtualserver-and-virtualserverroute-resources). The validation will be added in the future releases.

File: docs-web/configuration/policy-resource.md [hunk 8]
```diff
@@ -335,6 +341,77 @@ policies:
...
+     - No
+   * - ``protocols``
+     - Specifies the protocols for requests to an upstream HTTPS server. The default is ``TLSv1 TLSv1.1 TLSv1.2``.
+     - ``string`
+     - No
+```
+
+#### EgressMTLS Merging Behavior
+
+A VirtualServer/VirtualServerRoute can reference only a single EgressMTLS policy. Every subsequent reference will be ignored. For example, here we reference two policies:
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:407]: merging Behavior is similar to the JWT Merging Behavior, not IngressMTLS

File: docs-web/configuration/policy-resource.md [hunk 9]
```diff
@@ -335,6 +341,77 @@ policies:
...
+### EgressMTLS
+
+The EgressMTLS policy configures upstreams authentication and certificate verification.
+
+For example, the following policy will use `egress-mtls-secret` to authenticate with the upstream application and `egress-trusted-ca-secret` to verify the certificate of the application:
+```yaml
+egressMTLS:
+  tlsSecret: egress-mtls-secret
+  trustedCertSecret: egress-trusted-ca-secret
+  verifyServer: on
```
Comment (pleshakov) [inline docs-web/configuration/policy-resource.md:353]: I wonder if it make sense to use `true` here. For the boolean fields, we use `true|false`. Even for fields that correspond to directives that use `on|off`.   I think it is a bit confusing that we use `"on"` in IngressMTLS and `on` here, so I suggest using `true` here.

**Review by pleshakov — CHANGES_REQUESTED** [review]
there are a few problems, but overall looks good!

File: docs-web/configuration/policy-resource.md (hunk 4 above)
Comment (lucacome) [reply docs-web/configuration/policy-resource.md:376]: we don't have defaults for all the fields

File: docs-web/configuration/policy-resource.md (hunk 5 above)
Comment (lucacome) [reply docs-web/configuration/policy-resource.md:387]: we don't have defaults for all the fields

File: docs-web/configuration/policy-resource.md (hunk 6 above)
Comment (lucacome) [reply docs-web/configuration/policy-resource.md:391]: we don't have defaults for all the fields

**Review by pleshakov — APPROVED** [review]

**Pull request merged**